File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/RepositoryViewHolder.java
package com.kirchhoff.recyclermvp;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;


/**
 * @author Kirchhoff-
 */

public class RepositoryViewHolder extends RecyclerView.ViewHolder implements RepositoryRowView {

    TextView titleTextView;
    TextView startCountTextView;


    public RepositoryViewHolder(View itemView) {
        super(itemView);
        titleTextView = itemView.findViewById(R.id.repoTitleText);
        startCountTextView = itemView.findViewById(R.id.repoStartCountText);
    }


    @Override
    public void setTitle(String title) {
        titleTextView.setText(title);
    }

    @Override
    public void setStarCount(int starCount) {
        startCountTextView.setText(String.format("%s start", starCount));
    }
}

File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/RepositoryViewHolder.kt
package com.kirchhoff.recyclermvp

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class RepositoryViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView), RepositoryRowView {

    private val titleTextView: TextView = itemView.findViewById(R.id.repoTitleText)
    private val startCountTextView: TextView = itemView.findViewById(R.id.repoStartCountText)

    override fun setTitle(title: String) {
        titleTextView.text = title
    }

    override fun setStarCount(starCount: Int) {
        startCountTextView.text = String.format("%s start", starCount)
    }
}
